engine/StoryState.js

```diff
@@ -518,31 +549,16 @@ export class StoryState{
 	}
 	TrimNewlinesFromOutputStream(rightGlueToStopAt){
 		var removeWhitespaceFrom = -1;
-		var rightGluePos = -1;
-		var foundNonWhitespace = false;
 
-		// Work back from the end, and try to find the point where
-		// we need to start removing content. There are two ways:
-		//  - Start from the matching right-glue (because we just saw a left-glue)
-		//  - Simply work backwards to find the first newline in a string of whitespace
 		var i = this._outputStream.length-1;
 		while (i >= 0) {
 			var obj = this._outputStream[i];
-//			var cmd = obj as ControlCommand;
-			var cmd = obj;
-//			var txt = obj as StringValue;
-			var txt = obj;
-//			var glue = obj as Glue;
-			var glue = obj;
-
-			if (cmd instanceof ControlCommand || (txt instanceof StringValue && txt.isNonWhitespace)) {
-				foundNonWhitespace = true;
-				if( rightGlueToStopAt == null )
-					break;
-			} else if (rightGlueToStopAt && glue instanceof Glue && glue == rightGlueToStopAt) {
-				rightGluePos = i;
+			var cmd = (obj instanceof ControlCommand) ? obj : null;
```

Comment:
Instead of specifying the type in a comment above and performing the test in the condition, I perform the test inline and assign accordingly, which mimics what the C# code does with `as`.

Let me know if should revert the change!

Reply:
No that's actually way better, good idea 👍

Reply:
Nope that's actually way better, good idea 👍